app/assets/javascripts/discourse/app/services/sidebar-state.js

```diff
@@ -64,4 +66,12 @@ export default class SidebarState extends Service {
     this.panels = panels;
     this.mode = COMBINED_MODE;
   }
+
+  clearFilter() {
+    const filteredOutSections = this.filteredOutSections.toArray();
```

Comment:
Since you add/remove things from `filteredOutSections` in sidebar-section.js, which relies on this.filter for computation, you can actually just make this whole method only do:

```javascript
clearFilter() {
  this.filter = "";
}
```